bruig/flutterui/bruig/lib/components/usersearch/user_search_panel.dart

```diff
@@ -0,0 +1,319 @@
+import 'package:bruig/components/interactive_avatar.dart';
+import 'package:bruig/components/text.dart';
+import 'package:bruig/components/usersearch/user_search_model.dart';
+import 'package:bruig/models/client.dart';
+import 'package:bruig/models/emoji.dart';
+import 'package:bruig/models/uistate.dart';
+import 'package:bruig/screens/chats.dart';
+import 'package:bruig/screens/ln/components.dart';
+import 'package:bruig/theme_manager.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+
+class _SearchChatItemW extends StatefulWidget {
+  final ClientModel client;
+  final ChatModel chat;
+  final ValueChanged<ChatModel>? onChatTapped;
+  final UserSelectionModel? userSelModel;
+  const _SearchChatItemW(
+    this.client,
+    this.chat, {
+    this.onChatTapped,
+    this.userSelModel,
+  });
+
+  @override
+  State<_SearchChatItemW> createState() => __SearchChatItemWState();
+}
+
+class __SearchChatItemWState extends State<_SearchChatItemW> {
+  ClientModel get client => widget.client;
+  UserSelectionModel? get userSelModel => widget.userSelModel;
+  ChatModel get chat => widget.chat;
+  bool selected = false;
+
+  void onTap() {
+    if (widget.onChatTapped != null) {
+      widget.onChatTapped!(chat);
+    }
+    if (widget.userSelModel != null) {
+      setState(() => selected = widget.userSelModel!.toggle(chat));
+    }
+  }
+
+  void selModelChanged() {
+    var newSel = widget.userSelModel!.contains(chat);
+    if (newSel != selected) {
+      setState(() => selected = newSel);
+    }
+  }
+
+  @override
+  void initState() {
+    super.initState();
+    selected = userSelModel?.contains(chat) ?? false;
+    userSelModel?.addListener(selModelChanged);
+  }
+
+  @override
+  void didUpdateWidget(covariant _SearchChatItemW oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    selected = userSelModel?.contains(chat) ?? false;
+    if (userSelModel != oldWidget.userSelModel) {
+      oldWidget.userSelModel?.removeListener(selModelChanged);
+      userSelModel?.addListener(selModelChanged);
+    }
+  }
+
+  @override
+  void dispose() {
+    userSelModel?.removeListener(selModelChanged);
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return ListTile(
+      leading: UserMenuAvatar(client, chat),
```

Comment:
Was thinking we could not allow the Avatar click to show the floating sub menu here.